tests/beacon/test_helpers.py

```diff
@@ -0,0 +1,384 @@
+    (
+        'current_block_number,target_slot,success'
+    ),
+    [
+        (10, 0, True),
+        (10, 9, True),
+        (10, 10, False),
+        (128, 0, True),
+        (128, 127, True),
+        (128, 128, False),
+    ],
+)
+def test_get_block_hash(
+        genesis_block,
+        current_block_number,
+        target_slot,
+        success,
+        beacon_config):
+    cycle_length = beacon_config.cycle_length
+
+    blocks, recent_block_hashes = generate_mock_recent_block_hashes(
+        genesis_block,
+        current_block_number,
+        cycle_length,
+    )
+
+    if success:
+        block_hash = get_block_hash(
+            recent_block_hashes,
+            current_block_number,
+            target_slot,
+            cycle_length,
+        )
+        assert block_hash == blocks[target_slot].hash
+    else:
+        with pytest.raises(ValueError):
+            get_block_hash(
+                recent_block_hashes,
+                current_block_number,
+                target_slot,
+                cycle_length,
+            )
+
+
+@pytest.mark.parametrize(
+    (
+        'cycle_length,current_block_slot_number,from_slot,to_slot'
+    ),
+    [
+        (20, 10, 2, 7),
+        (20, 30, 10, 20),
+    ],
+)
+def test_get_hashes_from_recent_block_hashes(
+        genesis_block,
+        current_block_slot_number,
+        from_slot,
+        to_slot,
+        cycle_length):
+    _, recent_block_hashes = generate_mock_recent_block_hashes(
+        genesis_block,
+        current_block_slot_number,
+        cycle_length,
+    )
+
+    result = get_hashes_from_recent_block_hashes(
+        recent_block_hashes,
+        current_block_slot_number,
+        from_slot,
+        to_slot,
+        cycle_length,
+    )
+    assert len(result) == to_slot - from_slot + 1
+
+
+def test_get_hashes_to_sign(genesis_block, beacon_config):
+    cycle_length = beacon_config.cycle_length
+    current_block_slot_number = 1
+    blocks, recent_block_hashes = generate_mock_recent_block_hashes(
+        genesis_block,
+        current_block_slot_number,
+        cycle_length,
+    )
+
+    block = blocks[current_block_slot_number]
+    result = get_hashes_to_sign(
+        recent_block_hashes,
+        block,
+        cycle_length,
+    )
+    assert len(result) == cycle_length
+    assert result[-1] == blake(block.hash)
+
+
+def test_get_new_recent_block_hashes(genesis_block,
+                                     beacon_config,
+                                     sample_attestation_record_params):
+    cycle_length = beacon_config.cycle_length
+    current_block_slot_number = 15
+    blocks, recent_block_hashes = generate_mock_recent_block_hashes(
+        genesis_block,
+        current_block_slot_number,
+        cycle_length,
+    )
+
+    block = blocks[current_block_slot_number]
+    attestation = AttestationRecord(**sample_attestation_record_params)
+    oblique_parent_hashes = [b'\x77' * 32]
+    attestation = attestation.copy(
```

Comment:
*nitpick*

I tend to try and avoid overwriting variables.  If a variable *changes* during the scope of a function that introduces cognitive load to *track* what state it should be in.  Nothing structurally wrong with doing it but if we avoid it we end up with an overall easier to read codebase.

A pattern that you can use for this case would be to combine this with the line above to bet:

```python
oblique_parent_hashes = [b'\x77' * 32]
attestation = AttestationRecord(**sample_attestation_record_params).copy(
    slot=10,
    oblique_parent_hashes=oblique_parent_hashes,
)
```